functions/assignOfflineContact.ts

```diff
@@ -45,12 +56,19 @@ const cleanUpTask = async (task: TaskInstance, message: string) => {
 const assignToAvailableWorker = async (
   event: Body,
   newTask: TaskInstance,
-) => {
+  retry: number,
+): Promise<AssignmentResult> => {
   const reservations = await newTask.reservations().list();
   const reservation = reservations.find(r => r.workerSid === event.targetSid);
 
-  if (!reservation)
+  if (!reservation) {
+    if (retry < 3) {
```

Comment:
Since we don't know how long this might actually take, maybe increase this to 5?  Hopefully it doesn't take more than a second.